commit timer work

CHIRANJEET12 opened the pull request:
Added the timer feature

<img width="1035" height="835" alt="image" src="https://github.com/user-attachments/assets/1e52a8e4-1b4e-411a-a15b-8dd4a8e6f01a" />

CHIRANJEET12 commented:
commit timer work

CHIRANJEET12 commented:
commit timer work without any conflict

Priyakatariya commented:
@CHIRANJEET12 Resolve merge conflits and also follow below guidlines
If the PR template is not followed properly, I will not merge your pull request.

Please follow this PR template carefully:

📌 What does this PR do?
A brief summary of the changes you have made.

Related Issue
Fixes # (mention the issue number here)

🔧 Type of Change
 Bug fix (non-breaking change)

 New feature

 Breaking change

 Documentation update

✅ How Has This Been Tested?
Mention the steps for testing or an overview of the testing process.

Include environment details (Operating System, browser, Node version, etc.)

📸 Screenshots (if UI changed)
📊 Impact / Performance
Describe any performance changes or new dependencies introduced.

✅ Checklist
 I have followed the code style guidelines.

 I have self-reviewed my code.

 I have commented on complex parts of the code.

 I have updated or added documentation where necessary.

 All tests pass locally.

 No new warnings have been introduced.

CHIRANJEET12 commented:
Fixes timer functionality to properly pause/resume without resetting and ensures quiz inputs are disabled until start. Key changes include:
Added pause state management with isPaused and pausedTime
Removed accidental timer resets in start()
Standardized timer display updates
Improved input control flow
Fixed time's up message formatting

Type of Change
Bug fix (non-breaking change)
How Has This Been Tested?
Manually tested timer pausing/resuming at different intervals
Verified input disabling/enabling flow
Tested in Chrome/Brave

<img width="1035" height="835" alt="Screenshot 2025-07-22 225108" src="https://github.com/user-attachments/assets/361257f5-fa7d-460f-90cc-f51cba34782b" />

simple 2 buttons and a top Score and timer Bar added
📊 Impact / Performance
More accurate timer behavior
No new dependencies introduced
Slight memory optimization by removing redundant resets

✅ Checklist
Followed code style guidelines
Self-reviewed code
All existing tests pass
No new warnings

Priyakatariya commented:
@CHIRANJEET12  which issue number you solve . and your issue number is the number which issue you raise

CHIRANJEET12 commented:
#13 is the issue number.  #13
